presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -333,6 +336,26 @@ private boolean checkTablePermission(ConnectorSecurityContext context, SchemaTab
         return false;
     }
 
+    private boolean checkTablePermissionAny(ConnectorSecurityContext context, SchemaTableName tableName, TablePrivilege... anyPrivileges)
+    {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return true;
+        }
+
+        for (TableAccessControlRule rule : tableRules) {
+            Optional<Set<TablePrivilege>> tablePrivileges = rule.match(context.getIdentity().getUser(), tableName);
+            if (tablePrivileges.isPresent()) {
+                for (TablePrivilege anyPrivilege : anyPrivileges) {
+                    if (tablePrivileges.get().contains(anyPrivilege)) {
+                        return true;
+                    }
+                }
+                return false;
```

Comment:
this seems to be incorrect. Can you add a test where the second table rule actually gives any permission to a table?